From 2428c5f9f48f6a56ecb0e3c7a1c3348e2ef0e802 Mon Sep 17 00:00:00 2001
From: Gyuho Lee <leegyuho@amazon.com>
Date: Tue, 28 Sep 2021 19:38:47 +0000
Subject: [PATCH] --EKS-PATCH-- apiserver/healthz: upper log verbosity for
 "kms-provider-0"

--EKS-PRIVATE--

To silence

    cannot exclude some health checks, no health checks are installed
    matching "kms-provider-0"

This is logged when external health checker calls
"/healthz?exclude=kms-provider-0" against API server
that does not enable KMS encryption. Reduce such logs
to minimize the noise and log billing.

Signed-off-by: Gyuho Lee <leegyuho@amazon.com>
---
 .../src/k8s.io/apiserver/pkg/server/healthz/healthz.go    | 8 ++++++--
 .../k8s.io/apiserver/pkg/server/healthz/healthz_test.go   | 2 +-
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go b/staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go
index eb288239ce5..2beddcfb0a9 100644
--- a/staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go
+++ b/staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go
@@ -248,8 +248,12 @@ func handleRootHealth(name string, firstTimeHealthy func(), checks ...HealthChec
 			}
 		}
 		if excluded.Len() > 0 {
-			fmt.Fprintf(&individualCheckOutput, "warn: some health checks cannot be excluded: no matches for %s\n", formatQuoted(excluded.List()...))
-			klog.Warningf("cannot exclude some health checks, no health checks are installed matching %s",
+			// EKS-PATCH: to exclude "kms-provider-0" user-induced error, EKS has a separate alarm for encryption provider
+			verbosity := klog.V(6)
+			if verbosity.Enabled() {
+				fmt.Fprintf(&individualCheckOutput, "warn: some health checks cannot be excluded: no matches for %s\n", formatQuoted(excluded.List()...))
+			}
+			verbosity.Infof("cannot exclude some health checks, no health checks are installed matching %s",
 				formatQuoted(excluded.List()...))
 		}
 		// always be verbose on failure
diff --git a/staging/src/k8s.io/apiserver/pkg/server/healthz/healthz_test.go b/staging/src/k8s.io/apiserver/pkg/server/healthz/healthz_test.go
index 89937987e2c..42951e25434 100644
--- a/staging/src/k8s.io/apiserver/pkg/server/healthz/healthz_test.go
+++ b/staging/src/k8s.io/apiserver/pkg/server/healthz/healthz_test.go
@@ -106,7 +106,7 @@ func testMultipleChecks(path, name string, t *testing.T) {
 		{"?exclude=dontexist", "ok", http.StatusOK, false},
 		{"?exclude=bad", "ok", http.StatusOK, true},
 		{"?verbose=true&exclude=bad", fmt.Sprintf("[+]ping ok\n[+]bad excluded: ok\n%s check passed\n", name), http.StatusOK, true},
-		{"?verbose=true&exclude=dontexist", fmt.Sprintf("[+]ping ok\nwarn: some health checks cannot be excluded: no matches for \"dontexist\"\n%s check passed\n", name), http.StatusOK, false},
+		{"?verbose=true&exclude=dontexist", fmt.Sprintf("[+]ping ok\n%s check passed\n", name), http.StatusOK, false},
 		{"/ping", "ok", http.StatusOK, false},
 		{"", "ok", http.StatusOK, false},
 		{"?verbose", fmt.Sprintf("[+]ping ok\n[-]bad failed: reason withheld\n%s check failed\n", name), http.StatusInternalServerError, true},
